File: app/src/main/java/com/saitejajanjirala/imdbclone/ui/detail/DetailFragment.java
package com.saitejajanjirala.imdbclone.ui.detail;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bumptech.glide.Glide;
import com.google.android.flexbox.FlexDirection;
import com.google.android.flexbox.FlexboxLayoutManager;
import com.google.android.flexbox.JustifyContent;
import com.saitejajanjirala.imdbclone.R;
import com.saitejajanjirala.imdbclone.adapters.GenreAdapter;
import com.saitejajanjirala.imdbclone.databinding.FragmentDetailBinding;
import com.saitejajanjirala.imdbclone.models.Result;
import com.saitejajanjirala.imdbclone.utils.Utils;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;


public class DetailFragment extends Fragment {

    FragmentDetailBinding binding;
    Result result;
    ArrayList<String> genres;
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(false);
        genres=new ArrayList<>();
    }



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view= inflater.inflate(R.layout.fragment_detail, container, false);
        binding=FragmentDetailBinding.bind(view);
        result=DetailFragmentArgs.fromBundle(getArguments()).getResult();
        bindView();
        return binding.getRoot();
    }

    private void bindView() {
        if(result!=null){
            Utils.LoadImage(requireContext(),result.backdrop_path,binding.coverImage);
            Utils.LoadImage(requireContext(),result.poster_path,binding.posterImage);
            genres.addAll(Utils.getGenresFromIds(result.genre_ids));
            binding.rating.setText(result.vote_average+"");
            binding.title.setText(result.title);
            binding.topTitle.setText(result.title);
            binding.overView.setText(result.overview);
            binding.voters.setText(result.vote_count+"");
        }
        FlexboxLayoutManager layoutManager=new FlexboxLayoutManager(requireContext());
        layoutManager.setFlexDirection(FlexDirection.ROW);
        layoutManager.setJustifyContent(JustifyContent.FLEX_START);
        binding.generesRecyclerview.setLayoutManager(layoutManager);

        GenreAdapter adapter=new GenreAdapter(genres);
        binding.generesRecyclerview.setAdapter(adapter);
        binding.backButton.setOnClickListener(v -> closeFragment());
    }
    private void closeFragment(){
        Navigation.findNavController(binding.getRoot()).popBackStack();
    }


}

File: app/src/main/java/com/saitejajanjirala/imdbclonekotlin/ui/detail/DetailFragment.kt
package com.saitejajanjirala.imdbclonekotlin.ui.detail

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.Navigation
import com.google.android.flexbox.FlexDirection
import com.google.android.flexbox.FlexboxLayoutManager
import com.google.android.flexbox.JustifyContent
import com.saitejajanjirala.imdbclonekotlin.R
import com.saitejajanjirala.imdbclonekotlin.adapters.GenreAdapter
import com.saitejajanjirala.imdbclonekotlin.databinding.FragmentDetailBinding
import com.saitejajanjirala.imdbclonekotlin.models.Result
import com.saitejajanjirala.imdbclonekotlin.utils.Helper
import dagger.hilt.android.AndroidEntryPoint
import dagger.hilt.android.HiltAndroidApp
import java.util.ArrayList

@AndroidEntryPoint
class DetailFragment : Fragment() {

    private lateinit var binding:FragmentDetailBinding
    private lateinit var result:Result
    var genres: ArrayList<String>? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(false)
        genres = ArrayList()
    }


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_detail, container, false)
        binding = FragmentDetailBinding.bind(view)
        result = DetailFragmentArgs.fromBundle(requireArguments()).result
        bindView()
        return binding.root
    }

    private fun bindView() {
        if (result != null) {
            result.backdrop_path?.let { Helper.loadImage(requireContext(), it, binding.coverImage) }
            result.poster_path?.let { Helper.loadImage(requireContext(), it, binding.posterImage) }
            result.genre_ids?.let { Helper.getGenresFromIds(it)?.let { genres!!.addAll(it) } }
            binding.rating.text = result.vote_average.toString() + ""
            binding.title.text = result.title
            binding.topTitle.text = result.title
            binding.overView.text = result.overview
            binding.voters.text = result.vote_count.toString() + ""
        }
        val layoutManager = FlexboxLayoutManager(requireContext())
        layoutManager.flexDirection = FlexDirection.ROW
        layoutManager.justifyContent = JustifyContent.FLEX_START
        binding.generesRecyclerview.layoutManager = layoutManager
        val adapter = genres?.let { GenreAdapter(it) }
        binding.generesRecyclerview.adapter = adapter
        binding.backButton.setOnClickListener { closeFragment() }
    }

    private fun closeFragment() {
        Navigation.findNavController(binding.root).popBackStack()
    }
}